starknet_py/net/account/base_account.py

```diff
@@ -25,7 +28,45 @@
 from starknet_py.net.models.typed_data import TypedDataDict
 
 
-class BaseAccount(ABC):
+class OutsideExecutionSupportBaseMixin(ABC):
+
+    @abstractmethod
+    async def get_outside_execution_nonce(self) -> int:
+        """
+        Generate special valid nonce for outside execution calls.
+        """
+
+    @abstractmethod
+    async def supports_interface(
+        self, interface_id: OutsideExecutionInterfaceID
+    ) -> bool:
+        """
+        Check if the account supports the given outside execution interface. Part of ISRC5 standard.
+        """
+
+    @abstractmethod
+    async def sign_outside_execution_call(
```

Comment:
Applies to other occurrences
```suggestion
    async def sign_outside_transaction(
```